tuf/api/metadata.py

```diff
@@ -421,6 +421,13 @@ def verify_delegate(
 
             keys = self.signed.delegations.keys
             role = self.signed.delegations.roles.get(delegated_role)
+            if role is None:
+                for r in self.signed.delegations.roles.values():
+                    if (
+                        r.succinct_hash_info is not None
+                        and r.succinct_hash_info.is_bin(delegated_role)
+                    ):
+                        role = r
```

Comment:
I think this makes a good point for possibly simplifying the TAP: this is probably the best we can do with the current design but it really isn't obvious why we do this in the order we do and why we loop through all delegated roles

(the alternative implementation that avoids looping through all roles for the current TAP design would take `delegated_role`, try to parse it as a succinct delegation, translate "prefix-NNN" to "prefix-0", try to lookup "prefix-0" in `delegations.roles`, etc: still not simple or obvious)

Reply:
and if it's not obvious to the reader: the verifying code needs a way to find a succinct delegation based on a rolename, assuming the succinct delegation delegates to that role name... this is not trivial when there can be other, normal, delegations and multiple succinct delegations in the same Delegations structure.

Reply:
I will leave a comment to be a little clearer.